test/test_nestedtensor.py

```diff
@@ -0,0 +1,146 @@
+        a (int): the multiplier constant to the generator
+        c (int): the additive constant to the generator
+        m (int): the modulus constant to the generator
+    """
+    num_elements = 1
+    for s in size:
+        num_elements *= s
+
+    arr = [(a * seed + c) % m]
+    for i in range(num_elements - 1):
+        arr.append((a * arr[i] + c) % m)
+
+    return torch.tensor(arr).float().view(size) / m
+
+
+def random_int_tensor(seed, size, low=0, high=2 ** 32, a=22695477, c=1, m=2 ** 32):
+    """ Same as random_float_tensor but integers between [low, high)
+    """
+    return torch.floor(random_float_tensor(seed, size, a, c, m) * (high - low)) + low
+
+
+class TestNestedTensor(TestCase):
+
+    def gen_float_tensor(self, seed, shape):
+        return random_float_tensor(seed, shape)
+
+    def test_constructor(self):
+        tensors = []
+        for i in range(16):
+            tensors.append(self.gen_float_tensor(i, (16, 128, 128)))
+        nested_tensor = torch.nestedtensor(tensors)
+        for i in range(16):
+            tensors[i].mul_(i + 2)
+        for i in range(16):
+            assert (tensors[i] != nested_tensor.tensors[i]).all()
+
+    def test_nested_size(self):
+        a = torch.nestedtensor([torch.rand(1, 2), torch.rand(2, 3), torch.rand(4, 5)])
+        na = (torch.Size([1, 2]), torch.Size([2, 3]), torch.Size([4, 5]))
+        assert a.nested_size() == na
+
+
+    def test_len(self):
+        a = torch.nestedtensor([torch.tensor([1, 2]), 
+                                torch.tensor([3, 4]), 
+                                torch.tensor([5, 6]), 
+                                torch.tensor([7, 8])])
+        assert(len(a) == 4)
+        a = torch.nestedtensor([torch.tensor([1, 2]), 
+                                torch.tensor([7, 8])])
+
+        assert(len(a) == 2)
+        a = torch.nestedtensor([torch.tensor([1, 2])])
+        assert(len(a) == 1)
+        a = torch.nestedtensor([])
+        assert(len(a) == 0)
+
+
+    def test_unbind(self):
+        data = [self.gen_float_tensor(1, (2, 2)),
+                self.gen_float_tensor(2, (2, 2)),
+                self.gen_float_tensor(3, (2, 2))]
+        a = torch.nestedtensor(data)
+        b = a.unbind()
+        c = torch.nestedtensor([data_i + 1 for data_i in data])
+        for t in b:
+            t.add_(1)
+        assert (a == c).all()
+
+
+    def test_equal(self):
+        a1 = torch.nestedtensor([torch.tensor([1, 2]), 
+                                 torch.tensor([7, 8])])
+        a2 = torch.nestedtensor([torch.tensor([1, 2]), 
+                                 torch.tensor([7, 8])])
+        a3 = torch.nestedtensor([torch.tensor([3, 4]), 
+                                 torch.tensor([5, 6])])
+        # Just exercising them until we have __bool__, all() etc.
+        assert (a1 == a2).all()
+        assert (a1 != a3).all()
+        assert not (a1 != a2).any()
+        assert not (a1 == a3).any()
+
+
+    def test_unary(self):
```

Comment:
This only tests functionality, not coverage, but I suppose that'll come later.

Reply:
My goal here is to test the codegen, because I'm assuming the underlying functions (torch.cos etc.) are correct. So, I'm only exercising one function. I do agree, that in general this should be more rigorous, but at what point am I repeating test_torch etc. (this is not a rhetorical question).

Reply:
We should probably have a "tensor-type agnostic" version of test_torch so that we can run it on regular tensors and nested tensors (and possibly named tensors, but that might be a bit trickier). But yeah, I understand your point here; do you want to rename the test `test_unary_codegen`?